### PR TITLE
Add +powershell to lang/sh module

### PR DESCRIPTION
**Pull request by samwdp**
Added ability to get syntax highlighting on powershell scripts. Also added check for LSP to enable completion

### CODE REVIEW

**Review by hlissner — CHANGES_REQUESTED** [review]
Hi! Sorry for the late response. It seems like a lot of noise has crept into this PR. Would you mind rebasing it onto `develop` and making the marked changes?

**Review by hlissner — CHANGES_REQUESTED** [review]
Oh, and could you add a `+powershell` flag to `:lang sh` in docs/modules.org?

**Comment by samwdp** [comment]
I have made the changes in my fork but the changes are not reflected in the PR

File: modules/lang/sh/config.el [hunk 1]
```diff
@@ -74,8 +73,13 @@
...
 (use-package! fish-mode
   :when (featurep! +fish)
   :defer t
   :config (set-formatter! 'fish-mode #'fish_indent))
+
+(use-package! powershell
+	      :when (featurep! +powershell)
+	      :config
+	      (when (featurep! +lsp)
+		(add-hook 'powershell-mode-local-vars-hook #'lsp!)))
```
Comment (hlissner) [inline modules/lang/sh/config.el:85]: Please correct the indentation here. It should be 2 spaces (and 4 for the `add-hook` line).

File: modules/lang/sh/config.el [hunk 2]
```diff
@@ -74,8 +73,13 @@
...
   (setq company-shell-delete-duplicates t))
 
-
 (use-package! fish-mode
   :when (featurep! +fish)
   :defer t
   :config (set-formatter! 'fish-mode #'fish_indent))
+
+(use-package! powershell
+	      :when (featurep! +powershell)
```
Comment (hlissner) [inline modules/lang/sh/config.el:82]: Please add `:defer t` here, otherwise the package is eagerly loaded.

**Review by hlissner — CHANGES_REQUESTED** [review]

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Looks good to me. Thanks for your help!

**Pull request merged**